src/common/services/api/radioStations.service.js

```diff
@@ -0,0 +1,38 @@
+import angular from 'angular'
+import { Observable } from 'rxjs/Observable'
+import 'rxjs/add/operator/map'
+import 'rxjs/add/operator/mergeMap'
+import 'rxjs/add/operator/combineLatest'
+import 'rxjs/add/operator/pluck'
+import 'rxjs/add/observable/of'
+import 'rxjs/add/observable/throw'
+
+const serviceName = 'radioStationsService'
+
+class RadioStations {
+  /* @ngInject */
+  constructor ($http, $window, $log, $filter) {
+    this.$http = $http
+    this.sessionStorage = $window.sessionStorage
+    this.localStorage = $window.localStorage
+    this.$log = $log
+    this.$filter = $filter
+  }
+
+  getRadioStations(radioStationApiUrl, postalCode, radioStationRadius) {
+    return Observable.from(this.$http({
+      method: 'GET',
+      url: radioStationApiUrl + '/' + postalCode + '/' + radioStationRadius,
```

Comment:
Yeah I think it's possible for that to happen.  I wonder what a more robust solution would be.

Reply:
One option is to have a whitelist of domains or something.

Reply:
A whitelist could work... will we have to maintain that as more organizations use branded checkout?

Reply:
The maintenance would be specific to ministries who want to use radio stations, which should be very few.

Reply:
Are you thinking of implementing a whitelist on the API side, or should I whitelist things on the client side before a request is made?

Reply:
Hmm, to actually be secure it probably has to be on the server. Otherwise, I, as the malicious attacker could use local overrides in my Chrome dev tools to modify the whitelist to include my malicious api.

Reply:
Yeah I was thinking that.  That makes sense.  The whitelist should be in a secure location that only we would modify.

Reply:
I wonder; since the attacker could use dev tools to just remove the code that looks at the allowed list, are we really gaining any protection?

Reply:
What we do for TSYS device ID is that we have the ministry send in a keyword that is mapped to a real value on the server. Maybe we should do this instead of having them send in the API url itself?

Reply:
So, in this case, the server could build `radioStationApiUrl` that is shown here if you hit an endpoint and pass in an identifier (probably the same one that is passed for the tsys identifier, which would reduce what the ministries would need to configure and keep ids the same in the system)

Reply:
Hmm this is teaching me a lot about cybersecurity!  Yeah what you suggested at first makes it slightly more complicated to hack, but it's still not the most secure solution.  I see the benefit of having the URLs stored in the server and accessed through a keyword.  If we returned the URL from the server and used that, could it still get overridden?

Reply:
That's a good point. Any code on the front-end could be rewritten, including all of our calls to our own API. So maybe it's not worth the effort to use an allowed list at all...It's not too difficult to set up I suppose and it provides a little extra security. @mattdrees do you have an opinion on this discussion?

Reply:
I think I need a little more context.

> . Let's say a malicious agent builds out a branded checkout entering in a malicious URL here. Could they somehow pass data back to our server this way?

It seems to me that the malicious agent can enter zip codes, and get back radio stations. Which doesn't seem like a huge problem to me. Am I missing some bigger things that you're concerned about?

Reply:
The thought would be they could enter zip codes and get back anything, since they control the API (e.g. scripts for XSS, etc). However, as I think about it, an attacker can change any of the javascript code locally such that any api call can hit any api and return any data to be sent to our api, so maybe it's a moot point.

Reply:
So yeah, essentially this is a public api. There's nothing you can do to stop people from scraping your zip code-> radio database, if they want to. They could hire out people to collect it manually, and bypass any kind of captcha, even.

So I'd say don't bother with a whitelist.

The things you need to think about are whether a bad guy can do harm to a good user; somehow steal the info they type in or something. And if you have an XSS vuln, it's really hard to prevent that. Maybe a full CSP could. But a javascript-based solution wouldn't, because XSS by definition allows the attacker to muck up your javascript.